### ISSUE TITLE
Umbrella: Roadmap

### ISSUE THREAD

**Issue by jaredpalmer** [facebookexperimental/Recoil#428]
Based on my notes from my Zoom with @davidmccabe last week, here is my totally unofficial roadmap. Feel free to edit this or close it. 

**Core** 
- Internal cleanup (ongoing)
- Memory Leaks in 0.10 #366 
- Fast Refresh support #353 #12 #247
- Add internal FB playground/app to open source repo as example, codesandbox template for PR's #11 
- DevTools (FB intern was working on it, but currently stalled) #194 
- Server Rendering #131 #53 
- React Native #108 

**Docs**
- Atom Effects #384 
- A brief primer on normalized state
- "Thinking in Recoil" blog post (like "Thinking in React")
- Examples / Patterns
  - CRUD operations with REST API (not just reads)
  - Array operations (push, pop, insert, swap, etc)
  - Multiplayer
  - Undo/Redo
  - Change set

I know it's more work (given that a lot of it is duplicative), but I think it could help to move this into a full blog post or a public notion doc.

**Comment by drarmstr** [comment]
Yeah.. this is definitely unofficial.. For example, as we mentioned in our blog posts, we do have people working on concurrent mode, dev tools, an API for the singleton effects pattern for atom synchronization (not committing to Atom Effects quite yet), URL/localstorage persistence library, async fixes, other optimizations, etc.  We don't currently have people to drive SSR or React Native.  But, please remember, this library should still be considered "experimental," and I'm not sure we'd be comfortable publishing an "official roadmap" at this time.  Let us discuss.

**Comment by jlissner** [comment]
Any clues on when 0.0.11 will be released? I'm really interested in trying out Atom Effect (assuming you move forward with them)

**Comment by drarmstr** [comment]
> Any clues on when 0.0.11 will be released? I'm really interested in trying out Atom Effect (assuming you move forward with them)

We're not announcing a date yet for 0.0.11, but it's likely that Atom Effects will not be "published" with that release as we are still validating the API.  However, feel free to explore with the master branch.

**Comment by FuJuntao** [comment]
> > Any clues on when 0.0.11 will be released? I'm really interested in trying out Atom Effect (assuming you move forward with them)
> 
> We're not announcing a date yet for 0.0.11, but it's likely that Atom Effects will not be "published" with that release as we are still validating the API. However, feel free to explore with the master branch.

So can I assume that master branch is like a nightly build version?

**Comment by drarmstr** [comment]
Yes, the master branch is actually the live version that we use internally.

**Comment by vexii** [comment]
some ideas of CRUD Patterns would be nice 👍
